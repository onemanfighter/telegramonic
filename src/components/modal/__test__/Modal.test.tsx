import { fireEvent, render, screen } from '@testing-library/react';
import ModalComponent from '../Modal';

describe('Modal', () => {
  const onModalClose = jest.fn();

  beforeEach(() => {
    onModalClose.mockReset();
  });

  it('should render correctly if isOpen is true', () => {
    const { container } = render(
      <ModalComponent
        title="title"
        isOpen={true}
        onModalClose={onModalClose}
        message={'Message'}
      />,
    );

    expect(screen.findByText('title')).toBeDefined();
    expect(screen.findByText('Message')).toBeDefined();
  });

  it('should not render if isOpen is false', () => {
    const { container } = render(
      <ModalComponent
        title="title"
        isOpen={false}
        onModalClose={onModalClose}
        message={'Message'}
      />,
    );

    expect(screen.queryByText('title')).toBeNull();
    expect(screen.queryByText('Message')).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('should close modal on click of close button  ', () => {
    const { container } = render(
      <ModalComponent
        title="title"
        isOpen={true}
        onModalClose={onModalClose}
        message={'Message'}
      />,
    );

    fireEvent.click(screen.getByText('Close'));

    expect(onModalClose).toHaveBeenCalledTimes(1);
  });
});
