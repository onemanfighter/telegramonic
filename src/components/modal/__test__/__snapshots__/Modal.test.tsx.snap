// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should not render if isOpen is false 1`] = `<div />`;
